lib/filesystem/include/filesystem/filesystem.hpp

```diff
@@ -207,35 +207,137 @@ namespace irods::experimental::filesystem
+        ///                      must be an administrator to use this function.
+        /// \param[in] _comm     The communication object.
+        /// \param[in] _p        The path to a collection or data object.
+        /// \param[in] _metadata The metadata AVU to set.
+        ///
         /// \since 4.2.12
         auto set_metadata(admin_tag _admin, rxComm& _comm, const path& _p, const metadata& _metadata) -> void;
 
         auto add_metadata(rxComm& _comm, const path& _p, const metadata& _metadata) -> void;
 
+        /// \brief Adds a single metadata AVU on a collection or data object.
+        ///
+        /// \throws filesystem_error If the path is empty, exceeds the path limit, does not reference
+        ///                          a collection or data object, or an error occurs.
+        ///
+        /// \param[in] _admin    A tag which instructs the function to operate in administrator mode. The client
+        ///                      must be an administrator to use this function.
+        /// \param[in] _comm     The communication object.
+        /// \param[in] _p        The path to a collection or data object.
+        /// \param[in] _metadata The metadata AVU to add.
+        ///
         /// \since 4.2.12
         auto add_metadata(admin_tag _admin, rxComm& _comm, const path& _p, const metadata& _metadata) -> void;
 
         template <typename Iterator>
         auto add_metadata_atomic(rxComm& _comm, const path& _p, Iterator _first, Iterator _last) -> void;
 
+        /// \brief Atomically adds multiple metadata AVUs to a collection or data object.
+        ///
+        /// \throws filesystem_error If the path is empty, exceeds the path limit, does not reference
+        ///                          a collection or data object, or an error occurs.
+        ///
+        /// \param[in] _admin A tag which instructs the function to operate in administrator mode. The client
+        ///                   must be an administrator to use this function.
+        /// \param[in] _comm  The communication object.
+        /// \param[in] _p     The path to a collection or data object.
+        /// \param[in] _first An iterator pointing to the first element of a metadata AVU range.
+        /// \param[in] _last  An iterator pointing to the last element of a metadata AVU range (not inclusive).
+        ///
+        /// \since 4.2.12
+        template <typename Iterator>
+        auto add_metadata_atomic(admin_tag _admin, rxComm& _comm, const path& _p, Iterator _first, Iterator _last) -> void;
+
         template <typename Container,
                   typename = decltype(std::begin(std::declval<Container>())),
                   typename = std::enable_if_t<std::is_same_v<std::decay_t<typename Container::value_type>, metadata>>>
         auto add_metadata_atomic(rxComm& _comm, const path& _p, const Container& _container) -> void;
```

Comment:
Can we get docs on the rest of these functions? I realize these are just the non-admin versions, but just felt off to me. Perhaps they can be added in the earlier commit that introduced these functions. ...or they could all be squashed together. Up to you.

Reply:
sure. i'll document the other metadata functions.